File: java/src/main/java/parser/utils/ParseUtils.java
package parser.utils;

import parser.ast.ASTNode;

public class ParseUtils {

    /**
     * 后续排列验证
     * @param node
     * @return
     */
    public static String toPostfixExpression(ASTNode node){
        String leftStr ="";
        String rightStr = "";
        switch (node.getType()){
            case BINARY_EXPR:
                leftStr =toPostfixExpression(node.getChild(0));
                rightStr = toPostfixExpression(node.getChild(1));
                return leftStr+" "+rightStr+" " +node.getLexeme().getValue();
            case VARIABLE:
            case SCALAR:
                return node.getLexeme().getValue();
        }

        throw new RuntimeException("not implement");
    }
}

File: js/parser/util/ParseUtils.js
const ASTNodeType =require("../ast/ASTNodeTypes")

class ParseUtils {
    // postfix :后缀
    static  toPostfixExpression(astNode){
        switch (astNode.type) {
            case ASTNodeType.BINARY_EXPR:
                return `${ParseUtils.toPostfixExpression(astNode.getChild(0))} ${ParseUtils.toPostfixExpression(astNode.getChild(1))} ${ParseUtils.toPostfixExpression(astNode.lexeme.getValue())}`
            case ASTNodeType.VARIABLE:
            case ASTNodeType.SCALAR:
                return astNode.lexeme.getValue()
        }

        throw "not impl."
    }
}